src/adstaylorinteg.jl

```diff
@@ -0,0 +1,327 @@
+function split!(node::ADSBinaryNode{N, M, T}, p::SVector{M, Taylor1{TaylorN{T}}},
+                dt::T, nsplits::Int, maxsplits::Int, stol::T) where {N, M, T <: Real}
+    # Evaluate x at dt
+    x = _eval(p, dt)
+    # Split criteria for each element of x
+    mask = adsnorm.(x)
+    # Split
+    if nsplits < maxsplits && any(mask .> stol)
+        # Split
+        s1, x1, p1, s2, x2, p2 = split(node, x, p)
+        # Left half
+        leftchild!(node, s1, node.t + dt, x1, p1)
+        # Right half
+        rightchild!(node, s2, node.t + dt, x2, p2)
+        # Update number of splits
+        nsplits += 1
+    # No split
+    else
+        leftchild!(node, node.s, node.t + dt, x, p)
+    end
+
+    return nsplits
+end
+
+function taylorinteg(
+    f!, q0::SVector{M, TaylorN{T}}, s::ADSDomain{N, T}, t0::T, tmax::T,
+    order::Int, stol::T, abstol::T, params = nothing; maxsplits::Int = 10,
+    maxsteps::Int = 500, parse_eqs::Bool = true
+    ) where {N, M, T <: Real}
+
+    # Initialize the vector of Taylor1 expansions
+    t = t0 + Taylor1( T, order )
+    x = Array{Taylor1{TaylorN{T}}}(undef, M)
+    dx = Array{Taylor1{TaylorN{T}}}(undef, M)
+    @inbounds for i in eachindex(q0)
+        @inbounds x[i] = Taylor1( q0[i], order )
+        @inbounds dx[i] = Taylor1( zero(q0[i]), order )
+    end
+
+    # Determine if specialized jetcoeffs! method exists
+    parse_eqs, rv = _determine_parsing!(parse_eqs, f!, t, x, dx, params)
+
+    if parse_eqs
+        return _taylorinteg!(f!, q0, s, t0, tmax, order, stol, abstol, rv,
+                             params; maxsplits, maxsteps)
+    else
+        return _taylorinteg!(f!, q0, s, t0, tmax, order, stol, abstol,
+                             params; maxsplits, maxsteps)
+    end
+end
+
+function _taylorinteg!(
+    f!, q0::SVector{M, TaylorN{T}}, s::ADSDomain{N, T}, t0::T, tmax::T,
+    order::Int, stol::T, abstol::T, params = nothing; maxsplits::Int = 10,
+    maxsteps::Int = 500
+    ) where {N, M, T <: Real}
+
+    # Allocation
+    t = t0 + Taylor1( T, order )
+    dt = zeros(T, maxsplits)
+    x = Matrix{Taylor1{TaylorN{T}}}(undef, M, maxsplits)
+    dx = Matrix{Taylor1{TaylorN{T}}}(undef, M, maxsplits)
+    @inbounds for j in 1:maxsplits
+        @inbounds for i in eachindex(q0)
+            @inbounds x[i, j] = Taylor1( q0[i], order )
+            @inbounds dx[i, j] = Taylor1( zero(q0[i]), order )
+        end
+    end
+    nv = ADSBinaryNode{N, M, T}(s, t0, SVector{M, TaylorN{T}}(q0),
+                                SVector{M, Taylor1{TaylorN{T}}}(x[:, 1]))
+
+    nsteps = 1
+    nsplits = 1
+    sign_tstep = copysign(1, tmax-t0)
+
+    # Integration
+    while sign_tstep*t0 < sign_tstep*tmax
+
+        for (k, node) in enumerate(Leaves(nv))
+            @inbounds for i in 1:M
+                @inbounds x[i, k] = Taylor1( node.x[i], order )
+                @inbounds dx[i, k] = Taylor1( zero(node.x[i]), order )
+            end
+            xaux = Vector{Taylor1{TaylorN{T}}}(undef, M)
```

Comment:
I think it is better to define `xaux` outside the loop (perhaps as a matrix), and reuse it.

Reply:
Done.